### ISSUE TITLE
Handshake occurs on each connection

### ISSUE THREAD

**Issue by crockeea** [k0001/network-simple-tls#14]
When using `serve` and `connect`, repeated requests result in multiple handshakes, even when both the client and server settings have a session manager set. Why is TLS session not being resumed?

Code excerpts:
```
clientSettings :: Credential -> IO ClientSettings
clientSettings cred = do
    mgr <- newSessionManager defaultConfig
    let shared = doShared { sharedSessionManager = mgr, ...}
         cparams = (defaultParamsClient addr empty){ clientShared = shared }
         initSettings = makeClientSettings ("foo","bar") (Credentials [cred]) mempty
    return $ updateClientParams (const cparams) initSettings

serverSettings :: Credential -> IO ServerSettings
serverSettings (cert, _) = do
    mgr <- newSessionManager defaultConfig
    let shared = doShared { sharedSessionManager = mgr, ...}
          sParams = def { serverShared = shared }
          initSettings = makeServerSettings cred Nothing
    return $ updateServerParams (const sparams) initSettings

startServer cred = do
  sets<- serverSettings cred
  serve
        sets
        (Host ipaddr)
        (show port)
        (\(ctx,_) -> consume ctx)

clientSend cred msg = do
  sets <- clientSettings cred
  connect sets (Host destAddr) (show destPort) $ \(ctx,_) -> send ctx msg
```

Each call to `clientSend` results in a new TLS handshake.

**Comment by k0001** [comment]
@crockeea sorry I didn't reply before, but I missed the notification from GitHub.

I'll take a look at this as soon as possible.

**Comment by crockeea** [comment]
bump

**Comment by k0001** [comment]
Hi! I think there were some recent changes in the 'tls' library that may
help with this issue. I'll take a look as soon as possible :)

**Comment by k0001** [comment]
This has been implemented in `network-simple-tls-0.4`. 

Which, by the way, makes so minor but breaking changes to the API types https://hackage.haskell.org/package/network-simple-tls-0.4/changelog
